src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,53 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final Random random = new Random();
+    private final int maxSize = 15;
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int randomIndex = random.nextInt(5);
+        switch (randomIndex) {
+            case 0:
+                Circle circle = new Circle();
+                circle.setRadius(random.nextInt(maxSize));
+                circle.setColor(colorSupplier.getRandomColor());
+                return circle;
```

Comment:
create private methods (like getRandomCircle()) for all figures. And better use constructors, not setters.